hometasc-8

TVodovozov opened the pull request.

TVodovozov committed:
Переписана MainActivity с использованием библеотеки андройд-аннтотаций

TVodovozov committed:
Переписаны все фрагменты с использованием библеотеки аннотаций.
Добавлена библеотека ActiveAndroid.
Добавлена база данных Expenses и Category.
Переписан AddExpensesActivity.
Добавлен и реализован CategorySpinnerAdapter.
Добавлено новое activity MoneyTreckerApplication.
Добавлен и реализован

TVodovozov committed:
Реализован поиск по "Расходам".
Исправлена и добавлена реализация генерации Категорий в MainActivity.

TVodovozov committed:
Реализован поиск по Расходам с использование библиотеки аннотаций.
Исправлен баг отображения 2 значков поиска (лупа) в меню Поиска

TVodovozov committed:
Реализован поиск по Категориям с использование библиотеки аннотаций.

TVodovozov committed:
Реализована регистрация нового пользователя с помощью RegistrationActivity.
Создана форма для заполения регистрации нового пользователя.
Добавлены REST клиент, сервис, проверка статуса соединения.
Переписан SplashActivity.
Внесены необходимые изменения в Манифест.
Новые библиотеки добавлены в build нижнего уровня.

TVodovozov committed:
Исправлен баг в Main, когда при запуске приложения было пустое активити.
Добавлена регистрация нового пользователя.
Реализованы проверки подключения к интернету, проверка на минимальное кол-во символов, проверка статуса соединения.
При появлении статуса, что логин занят появляется сообщение в снекбаре.

KChernenko commented on app/src/main/java/com/loftschool/moneytracker/ui/RegistrationActivity.java:
```diff
@@ -54,37 +54,46 @@
-            e.printStackTrace();
+
+        if(!NetworkStatusChecker.isNetworkAvailable(this)) {
+            Snackbar.make(view, getString(R.string.registration_internet_error), Snackbar.LENGTH_SHORT).show();
```
В некоторых случаях получите исключение (приложение упадёт), потому что обращаетесь к UI из другого потока. Для решения этой проблемы нужно было использовать аннотацию @UiThread https://github.com/androidannotations/androidannotations/wiki/WorkingWithThreads#uithread

KChernenko commented on app/src/main/java/com/loftschool/moneytracker/ui/RegistrationActivity.java:
```diff
@@ -54,37 +54,46 @@
+                if (registrationModel.getStatus().equals(STATUS_SUCCESS)){
+                    MainActivity_.intent(this).start();
+                } else if (registrationModel.getStatus().equals(STATUS_LOGIN_BUSY_ALREADY)){
+                    Snackbar.make(view, getString(R.string.registration_server_status_busy_login_error), Snackbar.LENGTH_SHORT).show();
```
Аналогично можете получить исключение так как обращаетесь к UI из другого потока.

TVodovozov committed:
Исправлены ошибки пересечения потока запроса и UI.
Методы, помеченные специальной аннотацией  @UiThred, при возникновении ошибки регистрации выводят сообщение в снекбар.
Добавлена проверка пустого логина и пароля

TVodovozov committed:
Исправлено по замечаниям.
